src/pool/cli.c

```diff
@@ -1457,6 +1457,170 @@ dc_pool_query(tse_task_t *task)
 	return rc;
 }
 
+static int
+list_cont_bulk_create(crt_context_t ctx, crt_bulk_t *bulk,
+		      struct daos_pool_cont_info *buf, daos_size_t ncont)
+{
+	d_iov_t		iov;
+	d_sg_list_t	sgl;
+
+	d_iov_set(&iov, buf, ncont * sizeof(struct daos_pool_cont_info));
+	sgl.sg_nr = 1;
+	sgl.sg_nr_out = 0;
+	sgl.sg_iovs = &iov;
+
+	return crt_bulk_create(ctx, &sgl, CRT_BULK_RW, bulk);
+}
+
+static void
+list_cont_bulk_destroy(crt_bulk_t bulk)
+{
+	if (bulk != CRT_BULK_NULL)
+		crt_bulk_free(bulk);
+}
+
+struct pool_lc_arg {
+	crt_rpc_t			*rpc;
+	struct dc_pool			*lca_pool;
+	daos_size_t			 lca_req_ncont;
+	daos_size_t			*lca_ncont;
+	struct daos_pool_cont_info	*lca_cont_buf;
+};
+
+static int
+pool_list_cont_cb(tse_task_t *task, void *data)
+{
+	struct pool_lc_arg		*arg = (struct pool_lc_arg *) data;
+	struct pool_list_cont_in	*in = crt_req_get(arg->rpc);
+	struct pool_list_cont_out	*out = crt_reply_get(arg->rpc);
+	int				 rc = task->dt_result;
+
+	if (rc) {
+		D_ERROR("task error while listing containers: %d\n", rc);
+		D_GOTO(out, rc);
+	}
```

Comment:
This is not right. rc could be a retryable error and should go through pool_rsvc_client_complete_rpc first.

Reply:
Agree, will update in next push.